java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/deploy/mavenrepository/MavenRepositoryDeployer.java

```diff
@@ -0,0 +1,121 @@
+package com.redhat.hacbs.container.analyser.deploy.mavenrepository;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.deployment.DeployRequest;
+import org.eclipse.aether.deployment.DeploymentException;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.util.repository.AuthenticationBuilder;
+
+import io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext;
+import io.quarkus.bootstrap.resolver.maven.BootstrapMavenException;
+import io.quarkus.logging.Log;
+
+public class MavenRepositoryDeployer {
+    private final String username;
+
+    private final String password;
+
+    private final String repository;
+
+    private final Path artifacts;
+
+    private final RepositorySystem system;
+
+    private final DefaultRepositorySystemSession session;
+
+    public MavenRepositoryDeployer(BootstrapMavenContext mvnCtx, String username, String password, String repository,
+            Path artifacts)
+            throws BootstrapMavenException {
+        this.username = username;
+        this.password = password;
+        this.repository = repository;
+        this.artifacts = artifacts;
+
+        this.system = mvnCtx.getRepositorySystem();
+        this.session = MavenRepositorySystemUtils.newSession();
+
+        // TODO: Do we need a local repository manager?
+        session.setLocalRepositoryManager(system.newLocalRepositoryManager(session, new LocalRepository(artifacts.toFile())));
```

Comment:
I think a local manager is needed according to https://maven.apache.org/resolver/third-party-integrations.html